source/dotnet/Library/AdaptiveCards/AdaptiveCardConverter.cs

```diff
@@ -21,21 +21,25 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
         {
             var jObject = JObject.Load(reader);
 
-            if(jObject.Value<string>("type") != AdaptiveCard.TypeName)
+            if (jObject.Value<string>("type") != AdaptiveCard.TypeName)
                 throw new AdaptiveSerializationException($"Property 'type' must be '{AdaptiveCard.TypeName}'");
 
-            var typedElementConverter = serializer.ContractResolver.ResolveContract(typeof(AdaptiveTypedElement)).Converter;
+            if (reader.Depth == 0)
+            {
+                if (jObject.Value<string>("version") == null)
+                    throw new AdaptiveSerializationException("Could not parse required key: version. It was not found.");
 
-            var card = (AdaptiveCard)typedElementConverter.ReadJson(jObject.CreateReader(), objectType, existingValue, serializer);
+                // If this is the root AdaptiveCard and missing a version we fail parsing. 
+                // The depth checks that cards within a Action.ShowCard don't require the version
+                if (jObject.Value<string>("version") == "")
+                    throw new AdaptiveSerializationException("Property is required but was found empty: version");
 
-            // If this is the root AdaptiveCard and missing a version we fail parsing. 
-            // The depth checks that cards within a Action.ShowCard don't require the version
-            if (reader.Depth == 0 && card.Version == null)
-            {
-                // TODO: HACK for BF needing to deserialize legacy payloads that did not have a version
-                card.Version = "0.5";
```

Comment:
Do we no longer need this fallback?

Reply:
Yes, we need to be consistent across the the different platforms.